packages/x-data-grid/src/colDef/gridBooleanOperators.ts

```diff
@@ -6,7 +6,7 @@ export const getGridBooleanOperators = (): GridFilterOperator<any, boolean | nul
   {
     value: 'is',
     getApplyFilterFn: (filterItem: GridFilterItem) => {
-      if (!filterItem.value) {
+      if (!filterItem.value && filterItem.value !== false) {
```

Comment:
This should disable the filter when the value is `null`, `undefined` or `''` but not when it is `false`

Leaving support in for `''` and string values for the filterModel value because that is how the `<GridFilterInputBoolean />` used to set the value previously - don't want to break any user-land code that relied on this behaviour